server.go

```diff
@@ -400,12 +364,21 @@ func (s *Server) LoadNodeID() string {
 	return nodeID
 }
 
+type pilosaAddr URI
+
+func (p pilosaAddr) String() string {
+	uri := URI(p)
+	return uri.HostPort()
+
+}
+
+func (pilosaAddr) Network() string {
+	return "tcp"
+}
+
 // Addr returns the address of the listener.
 func (s *Server) Addr() net.Addr {
-	if s.ln == nil {
-		return nil
-	}
-	return s.ln.Addr()
+	return pilosaAddr(s.URI)
```

Comment:
little hack so that we can still return a net.Addr even though Server no longer has a net.Listener